core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -176,7 +170,7 @@ class Partition(val topic: String,
       // to maintain the decision maker controller's epoch in the zookeeper path
       controllerEpoch = partitionStateInfo.controllerEpoch
       // add replicas that are new
-      val newInSyncReplicas = partitionStateInfo.isr.asScala.map(r => getOrCreateReplica(r)).toSet
+      val newInSyncReplicas = partitionStateInfo.isr.asScala.map(r => getOrCreateReplica(r, partitionStateInfo.isNew)).toSet
```

Comment:
I think this is OK. `isNew` flag is only matters for local replica. And the local replica of the leader broker must be in the insync replicas set and it will be created properly in line 173.

Maybe I can add a comment here. Or do you like me to refactor the code a bit to call this method with the isNew flag?